libs/cms/src/lib/cms.module.ts

```diff
@@ -7,15 +7,24 @@ import {
   LatestNewsSliceResolver,
   FeaturedArticlesResolver,
   FeaturedSupportQNAsResolver,
+  PowerBiSliceResolver,
 } from './cms.resolver'
 import { CmsContentfulService } from './cms.contentful.service'
 import { ContentfulRepository } from './contentful.repository'
 import { CmsElasticsearchService } from './cms.elasticsearch.service'
 import { CmsHealthIndicator } from './cms.health'
 import { OrganizationLogoLoader } from './loaders/organizationLogo.loader'
+import { PowerBiServiceProvider } from './powerbi.service'
+import { ConfigModule } from '@island.is/nest/config'
+import { PowerBiConfig } from './powerbi.config'
 
 @Module({
-  imports: [TerminusModule],
+  imports: [
+    TerminusModule,
+    ConfigModule.forRoot({
+      load: [PowerBiConfig],
+    }),
```

Comment:
This should generally only be done in the app module, listing all config objects. But here you might have an issue since this is a large module which is imported in some other apps like the application system. Weird to configure powerbi there.

I can suggest two solutions:

1. Move the PowerBi config and resolver back to a powerbi domain, but don't expose any root level queries, just field resolvers like PowerBiSlice resolver. This domain/module is not imported in the application system.
2. Use `imports: [PowerBiConfig.registerOptional()],` here while also loading it in app.module.ts. Same pattern as [here](https://github.com/island-is/island.is/blob/9e53f8c204878c3a625e786ecfe33c35bff0470d/libs/clients/national-registry/v2/src/lib/nationalRegistryClient.module.ts#L10). It will be validated in api (because it's loaded in app.module.ts), but in application system, the config will have `config.isConfigured === false` and throw runtime exceptions if you try to read anything from it (be careful not to read anything from it at startup if `config.isConfigured === false`).

I kind of prefer #1 😅

Reply:
I'm struggling a lot with option #1 :/ 

The PowerBiSlice model is needed for the field resolver so I think I need to define that in the cms lib due to not having circular dependencies between the api domain and the cms (since the PowerBiSlice model is needed for the fields resolver)

Reply:
There is no need for the api domain so I'm hoping that what I've done most recently can work out 🙃